lib/call-cabal-project-to-nix.nix

```diff
@@ -212,13 +212,21 @@ let
                 (if inputMap.${repoData.url}.rev != repoData.ref
                   then throw "${inputMap.${repoData.url}.rev} may not match ${repoData.ref} for ${repoData.url} use \"${repoData.url}/${repoData.ref}\" as the inputMap key if ${repoData.ref} is a branch or tag that points to ${inputMap.${repoData.url}.rev}."
                   else inputMap.${repoData.url})
-            else if repoData.sha256 != null
+            else if repoData.sha256 != null && !repoData.is-private
```

Comment:
So if there's no `sha256` and the repo isn't private, then we do a regular `fetchgit`.

Reply:
So why does the privacy matter here? If we're setting `ref` or `rev` at the source, then it seems to me like these two branches are equivalent.

Reply:
wow, fascinating. thanks for the explanation!